microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py

```diff
@@ -148,13 +217,64 @@ def _send_subtest_msg(
         environment: Environment,
         test_name: str,
         test_status: TestStatus,
+        test_message: str = "",
     ) -> None:
         subtest_msg = create_test_result_message(
-            SubTestMessage,
-            test_id,
-            environment,
-            test_name,
-            test_status,
+            SubTestMessage, test_id, environment, test_name, test_status, test_message
         )
 
         notifier.notify(subtest_msg)
+
+    def _list_perf_metrics_tests(self, hypervisor="kvm") -> List[str]:
+
+        self._log.debug("Listing the performance test cases")
+        tests_list = []
+        result = self.run(
+            f"tests --hypervisor {hypervisor} --metrics -- -- --list-tests",
+            timeout=self.TIME_OUT,
+            force_run=True,
+            cwd=self.repo_root,
+            shell=True,
+            expected_exit_code=0,
+        )
+
+        stdout = result.stdout
+
+        # Ex. String for below regex : "boot_time_ms" (test_timeout = 2s, test_iterations = 10)
+        regex = '\\"(.*)\\" \\('
+
+        pattern = re.compile(regex)
+        tests_list = pattern.findall(stdout)
+
+        self._log.debug(f"Testcases found: {tests_list}")
+        return tests_list
+
+    def _process_perf_metric_test_result(self, output: str) -> str:
+        cnt = 0
+        for line in output.split("\n"):
+            if line.find("git_human_readable") >= 0:
+                cnt -= 1
+                break
+            cnt += 1
+
+        result = "\n".join([i.strip() for i in output.split("\n")[cnt:-5]])
+        self._log.debug(f"Result from testcase stdout : {result}")
```

Comment:
Use the perf message instead of dump results in a separated file, because the notifier is extendable. The perf message can help collecting environment information too.

Reply:
If it needs to consume json format result, you can add a json notifier to consume test results.

Reply:
Yes, I am using notifier to push the testcase result. This function is just to extract the metrics data if testcase is successful which is part of stdout.

Reply:
Use notifier to dump it to file. So it's configurable, and avoid inconsistent result handling ways across test suites.

Reply:
Sure i will remove dumping result in json file.